File: api/src/main/java/com/zeven/ets_proyect/Ets_Proyect/dto/SneakerDTO.java
package com.zeven.ets_proyect.Ets_Proyect.dto;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class SneakerDTO {

    @JsonProperty("id_product")
    private Long originId;

    @JsonProperty("id_collection")
    private Long originCollectionId;

    @JsonProperty("available_sizes")
    private List<AvailableSizesDTO> size;

    @JsonProperty("image_alt")
    private String imageAlt;

    @JsonProperty("tax_rate")
    private String taxRate;

    @JsonProperty("original_price")
    private Integer originalPrice;

    @JsonProperty("price_base")
    private Double priceBase;

    @JsonProperty("tax_amount")
    private Double taxAmount;

    private Integer price;
    private String brand;
    private String image;
    private String link;
    private String name;
    private String reference;
}

File: app/src/app/dto/sneaker-dto.ts
import SneakerInterface from "../interface/sneaker-interface";

export default class SneakerDTO implements SneakerInterface{
    sneakerId: number;
    originId: number;
    originCollectionId: number;
    imageAlt: string;
    taxRate: string;
    originalPrice: number;
    priceBase: number;
    taxAmoun: number;
    price: number;
    brand: string;
    image: string;
    link: string;
    name: string;
    reference: string;

    constructor (data: Partial<SneakerInterface>) {
        this.sneakerId = data.sneakerId ?? 0;
        this.originId = data.originId ?? 0;
        this.originCollectionId = data.originCollectionId ?? 0;
        this.imageAlt = data.imageAlt ?? '';
        this.taxRate = data.taxRate ?? '';
        this.originalPrice = data.originalPrice ?? 0;
        this.priceBase = data.priceBase ?? 0;
        this.taxAmoun = data.taxAmoun ?? 0;
        this.price = data.price ?? 0;
        this.brand = data.brand ?? '';
        this.image = data.image ?? '';
        this.link = data.link ?? '';
        this.name = data.name ?? '';
        this.reference = data.reference ?? '';
    }
}